### PR TITLE
SUPNXP-18275 Define configurable fulltextFieldSizeLimit

### PR DESCRIPTION
**Pull request by ron1**

### CODE REVIEW

File: nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/main/java/org/nuxeo/ecm/core/storage/sql/RepositoryDescriptor.java [hunk 1]
```diff
@@ -310,6 +310,11 @@ public FulltextDescriptor getFulltextDescriptor() {
         return fulltextDescriptor;
     }
 
+    @XNode("indexing/fulltext@fieldSizeLimit")
+    public void setFulltextFieldSizeLimit(int fieldSizeLimit) {
+        fulltextDescriptor.setFulltextFieldSizeLimit(fieldSizeLimit);
+    }
+
```
Comment (efge) [inline nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/main/java/org/nuxeo/ecm/core/storage/sql/RepositoryDescriptor.java]: Please add similar changes to `DBSRepositoryDescriptor` using ` @XNode("fulltext@fieldSizeLimit")`.

File: nuxeo-core/nuxeo-core-storage/src/main/java/org/nuxeo/ecm/core/storage/FulltextExtractorWork.java [hunk 2]
```diff
@@ -148,6 +148,15 @@ protected void extractBinaryText() {
...
             List<Blob> blobs = extractor.getBlobs(doc);
             String text = blobsToText(blobs, docId);
+            int fullTextFieldSizeLimit = fulltextConfiguration.fulltextFieldSizeLimit;
+            if (fullTextFieldSizeLimit != 0 && text.length() > fullTextFieldSizeLimit) {
+                if (log.isDebugEnabled()) {
+                    log.debug(String.format(
+                            "Fulltext extraction of length: %s for indexName: %s of document: %s too large to process",
+                            docId, text.length()));
+                }
+                continue;
```
Comment (efge) [inline nuxeo-core/nuxeo-core-storage/src/main/java/org/nuxeo/ecm/core/storage/FulltextExtractorWork.java]: Why not just truncate to the limit?

File: nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestSQLBinariesIndexingOverride.java [hunk 3]
```diff
@@ -186,4 +186,28 @@ public void testExcludeFieldBlob() throws Exception {
         assertEquals(0, res.size());
     }
 
+    @Test
+    public void testEnforceFulltextFieldSizeLimit() throws Exception {
+        DocumentModelList res;
+        for (int i = 0; i < 2; i++) {
+            String namePrefix = i == 0 ? "reg" : "big";
```
Comment (efge) [inline nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestSQLBinariesIndexingOverride.java]: I'd do `boolean regular = i == 0` and use this instead (in two places) for clarity.

File: nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestSQLBinariesIndexingOverride.java [hunk 4]
```diff
@@ -186,4 +186,28 @@ public void testExcludeFieldBlob() throws Exception {
...
+            }
+            DocumentModel doc = session.createDocumentModel("/", name, "File");
+            doc.setPropertyValue("file:content", (Serializable) Blobs.createBlob(content));
+            doc = session.createDocument(doc);
+            session.save();
+
+            waitForFulltextIndexing();
+
+            // main index
+            res = session.query("SELECT * FROM Document WHERE ecm:fulltext = '" + content + "'");
```
Comment (efge) [inline nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestSQLBinariesIndexingOverride.java]: Even in unit tests it's good to be exemplar and use `NXQL.escapeString` when creating NXQL.

File: nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestSQLBinariesIndexingOverride.java [hunk 5]
```diff
@@ -186,4 +186,28 @@ public void testExcludeFieldBlob() throws Exception {
...
+    public void testEnforceFulltextFieldSizeLimit() throws Exception {
+        DocumentModelList res;
+        for (int i = 0; i < 2; i++) {
+            String namePrefix = i == 0 ? "reg" : "big";
+            String name = namePrefix + "Content";
+            // fulltextFieldSizeLimit configured as 1024
+            String content = name + " ";
+            for (int j = 0; j < (i * 100) + 50; j++) {
+                content += name + " ";
+            }
```
Comment (efge) [inline nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestSQLBinariesIndexingOverride.java]: I would add assertions on `content.size()` depending on i to make that explicit.

**Comment by ron1** [comment]
Replaced by https://github.com/nuxeo/nuxeo/pull/307.

**Pull request closed without merging**